src/openforms/submissions/api/viewsets.py

```diff
@@ -297,25 +296,61 @@ def _complete(self, request, *args, **kwargs):
         status endpoint that a retry is needed, the ID is added back to the session.
 
         ---
-        **Warning**
+        **Validation errors**
+
+        The validation errors are returned in the usual `invalidParams` structure. The
+        following errors can occur:
+
+        * name `privacyPolicyAccepted` - unchecked while accepting it is required
+        * name `statementOfTruthAccepted` - unchecked while accepting it is required
+        * name `steps[i].nonFieldErrors`:
+
+            * code `blocked` - a logic rule prevents the step from being submitted,
+              which blocks the submission as a whole from being completed.
+            * code `incomplete` - there is no or incomplete step data submitted for the
+              step at this index.
+
+        * name `steps[i].data.*` - validation errors related to the formio components
+          in the step data.
+
+        Additionally, if you try to complete a submission that doesn't allow this (due
+        to form-level configuration), you will get an HTTP 403 error.
 
-        The schema of the validation errors response is currently marked as
-        experimental. See our versioning policy in the developer documentation for
-        what this means.
         ---
         """
         submission = self.get_object()
+        if submission.form.submission_allowed != SubmissionAllowedChoices.yes:
+            raise PermissionDenied(
+                detail=_("Submission is not enabled for this form."),
+                code="submission-not-allowed",
+            )
+
+        # prepare the submission with evaluated logic so that its state can be validated
+        submission.load_execution_state()
```

Comment:
Do we need this here?The state is loaded again in the `check_submission_logic` (checking because I see a lot of loops inside the method so maybe we could avoid doing this twice)

Reply:
I prefer to make it explicit - loading the state is idempotent, if it's already cached, it will not load it again :)

Reply:
Perfect, missed that it is cached